nsds_lab_to_nwb/components/tdt/tdt_manager.py

```diff
@@ -34,5 +35,31 @@ def extract_tdt(self, device_name, dev_conf, electrode_table_region):
 
     def __extract_from_tdt_files(self, device_name, dev_conf):
         tdt_path = self.tdt_path
-        # return data
-        raise NotImplementedError('TODO')
+        header = tdt.read_block(tdt_path, headers=1)
+        num_channels = len(header.stores[device_name].chan)
+        all_tdt_data = tdt.read_block(tdt_path) #I see no way to individually pull data streams using tdt's package, inefficent!
```

Comment:
What's the type of `all_tdt_data` or maybe `all_tdt_data.streams[device_name]` and `all_tdt_data.streams[device_name].data`? It would be helpful to know when it gets read into memory. If `tdt` only supports reading the whole file into memory at once, there's probably no reason to worry about iterative write.

Reply:
It's possible that one of these objects exposes something like memory mapped access. Maybe you can load channels one at a time?
https://www.tdt.com/docs/sdk/offline-data-analysis/offline-data-python/overview/
If that works without loading all channels, then iterative write could work.

Reply:
@JesseLivezey the tdt package supports loading individual channels or segments of the data. Actually my first approach was to create a generator that return a numpy array of one channel at a time, but it fails when it comes time to dump write the nwb file to storage. The error is this weird broadcast error that I couldn't figure out. I'm not sure if the `'all_tdt_data` returns a memory mapped thing. I believe the data type was a struct and that's the same for `all_tdt_data.streams[device_name]`. xxxx.data is a numpy array. Any help on how to overcome this broadcast error would be great. @VBaratham any advice for how to get iterative write to nwb files to work? I can copy paste the specific error later tonight.